lib/vet/tarball.js

```diff
@@ -9,16 +9,24 @@ module.exports = function(module, dir, url, json) {
   var moduleDir = path.join(dir, 'node_modules', module);
   return new Promise(function(resolve, reject) {
     var children = [];
-    var stream = got.stream(url).pipe(zlib.Unzip()).pipe(tar.Parse());
+    var stream = got.stream(url).pipe(zlib.Unzip()).pipe(new tar.Parse());
     stream.on('entry', function(arg) {
-      if (arg.type !== 'File') return;
+      if (arg.type !== 'File') { arg.resume(); return; }
       var parts = arg.path.split('/');
-      if (arg.path.indexOf('node_modules') !== -1) return;
-      if (parts[1] === 'package.json') return children.push(vetPackageJSON(moduleDir, arg));
-      if (parts[parts.length-1] === '.gitignore') return;
-      if (parts[parts.length-1].slice(-3) !== '.js') return;
+      if (arg.path.indexOf('node_modules') !== -1) { arg.resume(); return; }
```

Comment:
what does `arg.resume()` do?

Reply:
This is what I found from [minipass used by tar](https://github.com/isaacs/minipass)
resume() - Resume the stream. If there's data in the buffer, it is all discarded. Any buffered events are immediately emitted. 

I don't fully understand how it works. What happen was that, when you don't process the data and just return, it seems to just pause and never emits the end and hangs there. So then I stumble upon some people using the tar with resume and I tried it and it worked to pass the tests.